test/EndToEndTests/Tests/Core/Microsoft.OData.Core.E2E.Tests/WriteJsonPayloadTests/SerializationInfoEdgeTests.cs

```diff
@@ -0,0 +1,488 @@
+        /// </summary>
+        [Theory]
+        [InlineData(MimeTypeODataParameterFullMetadata)]
+        [InlineData(MimeTypeODataParameterMinimalMetadata)]
+        [InlineData(MimeTypeODataParameterNoMetadata)]
+        public async Task SpecifySerializationInfoForFeedAndEntry(string mimeType)
+        {
+            var settings = new ODataMessageWriterSettings();
+            settings.ODataUri = new ODataUri() { ServiceRoot = _baseUri };
+
+            var responseMessageWithoutModel = new TestStreamResponseMessage(new MemoryStream());
+            responseMessageWithoutModel.SetHeader("Content-Type", mimeType);
+            using (var messageWriter = new ODataMessageWriter(responseMessageWithoutModel, settings))
+            {
+                var odataWriter = await messageWriter.CreateODataResourceSetWriterAsync();
+                var feed = this.CreatePersonFeed();
+                feed.SetSerializationInfo(new ODataResourceSerializationInfo() { NavigationSourceName = "People", NavigationSourceEntityTypeName = NameSpacePrefix + "Person" });
+
+                var entry = new ODataResource()
+                {
+                    Id = new Uri(_baseUri + "People(-5)"),
+                    TypeName = NameSpacePrefix + "Employee"
+                };
+
+                var personEntryP1 = new ODataProperty { Name = "PersonId", Value = -5 };
+                var personEntryP2 = new ODataProperty
+                {
+                    Name = "Name",
+                    Value = "xhsdckkeqzvlnprheujeycqrglfehtdocildrequohlffazfgtvmddyqsaxrojqxrsckohrakdxlrghgmzqnyruzu"
+                };
+
+                var personEntryP3 = new ODataProperty { Name = "ManagersPersonId", Value = -465010984 };
+
+                entry.Properties = new[] { personEntryP1, personEntryP2, personEntryP3 };
+                entry.SetSerializationInfo(new ODataResourceSerializationInfo() { NavigationSourceName = "Person", NavigationSourceEntityTypeName = NameSpacePrefix + "Person" });
+
+                await odataWriter.WriteStartAsync(feed);
+                await odataWriter.WriteStartAsync(entry);
+                await odataWriter.WriteEndAsync();
+                await odataWriter.WriteEndAsync();
+                Stream stream = await responseMessageWithoutModel.GetStreamAsync();
+                string result = await TestsHelper.ReadStreamContentAsync(stream);
+                if (!mimeType.Contains(MimeTypes.ODataParameterNoMetadata))
+                {
+                    Assert.Contains(NameSpacePrefix + "Employee", result);
+                    Assert.Contains("$metadata#People", result);
+                }
+                else
+                {
+                    Assert.DoesNotContain(NameSpacePrefix + "Employee", result);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Write response payload with serialization info containing wrong values

```

Comment:
What is the behaviour being tested? What is the expected outcome?

Reply:
I've update the test